src/Composer/ScriptHandler.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Composer;
+
+use Composer\Script\Event;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class ScriptHandler
+{
+    const SWAGGER_UI_DESTINATION = 'web/swagger-ui';
+    const SWAGGER_UI_SOURCE = 'vendor/swagger-api/swagger-ui/dist';
+
+    /**
+     * Copy swagger ui to the correct directory.
+     */
+    public static function installSwaggerUi()
+    {
+        $fs = new Filesystem();
+        $cwd = getcwd();
+
+        foreach ((new Finder())->in(implode('/', [$cwd, self::SWAGGER_UI_SOURCE])) as $file) {
```

Comment:
What do you think about something like that?
```php
class ScriptHandler
{
    protected static $symfonyWebDir = 'web';

    public static function installSwaggerUi(Event $event)
    {
        $vendorDir = $event->getComposer()->getConfig()->get('vendor-dir');
        $webDir = $event->getComposer()->getPackage()->getExtra()['symfony-web-dir'] ?? static::symfonyWebDir;

        $event->getIO()->write('Trying to install assets for Swagger-UI...');

        (new Filesystem())->mirror(
            $webDir.'/swagger-api/swagger-ui/dist',
            $vendorDir.'/swagger-ui',
            null,
            ['override' => true, 'delete' => true]
        );
    }
}
```
It should work, isn't it?

cc @dunglas @teohhanhui